File: src/main/java/org/dxworks/githubminer/dto/export/CommitInfoDTO.java
package org.dxworks.githubminer.dto.export;

import com.google.api.client.util.Key;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.dxworks.githubminer.dto.response.repository.commits.RepoCommit;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CommitInfoDTO {
	@Key
	private String id;
	@Key
	private UserDTO author;
	@Key
	private UserDTO committer;

	public static CommitInfoDTO fromCommit(RepoCommit repoCommit) {
		if (repoCommit == null || com.google.api.client.util.Data.isNull(repoCommit))
			return null;
		return builder()
				.id(repoCommit.getSha())
				.author(UserDTO.fromUser(repoCommit.getAuthor()))
				.committer(UserDTO.fromUser(repoCommit.getCommitter()))
				.build();
	}
}

File: github-miner-app/src/main/kotlin/org/dxworks/githubminer/dto/export/CommitInfoDTO.kt
package org.dxworks.githubminer.dto.export

import com.google.api.client.util.Data
import com.google.api.client.util.Key
import org.dxworks.githubminer.dto.response.repository.commits.RepoCommit

class CommitInfoDTO {
    @Key
    var id: String? = null

    @Key
    var author: UserDTO? = null

    @Key
    var committer: UserDTO? = null

    companion object {
        fun fromCommit(repoCommit: RepoCommit?): CommitInfoDTO? {
            return if (repoCommit == null || Data.isNull(repoCommit)) null else CommitInfoDTO().apply {
                id = repoCommit.sha
                author = UserDTO.fromUser(repoCommit.author)
                committer = UserDTO.fromUser(repoCommit.committer)
            }
        }
    }
}
